mender-convert

```diff
@@ -266,9 +263,21 @@ do_raw_disk_image_create_partitions() {
 
   mkdir -p $output_dir && cd $output_dir
 
-  # In case of missing .sdimg name use the default format.
-  [ -z $mender_disk_image ] && mender_disk_image=$output_dir/mender_${device_type}.sdimg \
-                            || mender_disk_image=$output_dir/$mender_disk_image
+  # Make sure the user's given Mender image name has a correct extension.
+  # If Mender image name is not provided, then use following syntax:
+  # mender-<device_name>-<artifact_name>.sdimg
+  if [ -n "${mender_disk_image}" ]; then
+    mender_disk_basename=$(basename -- "$mender_disk_image")
+    if [[ $mender_disk_basename =~ \.sdimg$ ]]; then
+      mender_disk_image=$output_dir/$mender_disk_basename
+    else
+      mender_disk_filename="${mender_disk_basename%.*}"
+      mender_disk_image=$output_dir/${mender_disk_filename}.sdimg
+    fi
+  else
+    [ -n "$artifact_name" ] \
```

Comment:
You already have check (-z) on `artifact_name` at the beginning of the function, meaning that this check (-n) is not really needed.